s/bindings/parse.test.ts

```diff
@@ -4,123 +4,92 @@ import {Suite, expect} from "cynic"
 import {Binds} from "./types.js"
 import {emojis} from "./emojis.js"
 import {parseBindings} from "./parse.js"
+import {runTheseTestsOnEachBindType} from "./parsing/run-these-tests-on-each-bind-type.js"
 
 export default <Suite>{
 
-	"action binds": async() => {
-		const testChannel = (channel: keyof typeof emojis) => ({
-
-			"bind with no content": async() => {
-				const emoji = emojis[channel]
-				const binds = <Binds>parseBindings(`
-					${emoji} Benevolent ::
-				`)[channel]
-				expect(Object.keys(binds).length).equals(1)
-			},
-
-			"single bind": async() => {
-				const emoji = emojis[channel]
-				const binds = <Binds>parseBindings(`
-					${emoji} Benevolent :: Alpha
-				`)[channel]
-				expect(Object.keys(binds).length).equals(1)
-				const [a] = binds["Benevolent"] ?? []
-				expect(a).equals("Alpha")
-			},
-
-			"two binds": async() => {
-				const emoji = emojis[channel]
-				const binds = <Binds>parseBindings(`
-					${emoji} Benevolent :: Alpha Bravo
-				`)[channel]
-				const [a, b] = binds["Benevolent"] ?? []
-				expect(a).equals("Alpha")
-				expect(b).equals("Bravo")
-			},
-
-			"two binds with weird whitespace": async() => {
-				const emoji = emojis[channel]
-				const binds = <Binds>parseBindings(`
-					${emoji}
-						Benevolent
-							::
-							Alpha  Bravo  
-				`)[channel]
-				const [a, b] = binds["Benevolent"] ?? []
-				expect(a).equals("Alpha")
-				expect(b).equals("Bravo")
-			},
-		})
-
-		return {
-			key: testChannel("key"),
-			mouse: testChannel("mouse"),
-			vector2: testChannel("vector2"),
-		}
-	},
+	"action binds": runTheseTestsOnEachBindType(channel => ({
+
+		"bind with no content": async() => {
+			const emoji = emojis[channel]
+			const binds = <Binds>parseBindings(`
+				${emoji} Benevolent :: `)[channel]
+			expect(Object.keys(binds).length).equals(1) },
+
+		"single bind": async() => {
+			const emoji = emojis[channel]
+			const binds = <Binds>parseBindings(`
+				${emoji} Benevolent :: Alpha `)[channel]
+			expect(Object.keys(binds).length).equals(1)
+			const [a] = binds["Benevolent"] ?? []
+			expect(a).equals("Alpha") },
+
+		"two binds": async() => {
+			const emoji = emojis[channel]
+			const binds = <Binds>parseBindings(`
+				${emoji} Benevolent :: Alpha Bravo `)[channel]
+			const [a, b] = binds["Benevolent"] ?? []
+			expect(a).equals("Alpha")
+			expect(b).equals("Bravo") },
+
```

Comment:
i'm experimenting with this weird syntax,

where i'm trying to "make it read like python"

i call it "tucked" style, and i think i might like it for test suites i think...

what do you think?

Reply:
Whats the difference between those styles ? If you like it I will try do adapt whenever I will write any test